### ISSUE TITLE
makedns generates broken named.conf on DNS slave

### ISSUE THREAD

**Issue by kcgthb** [xcat2/xcat-core#4392]
`makedns -s` generates a `/etc/named.conf` file on DNS slaves that looks like this (on CentOS 7.x, xCAT Version: 2.13.8):
```
#generated by xCAT: /opt/xcat/sbin/makedns command
options {
    forwarders {
        8.8.8.8;
    };
    allow-transfer { any; };
};

zone "...
```
It misses the `directory "/var/named";` directive, which is required for proper operation. Without it, `named` tries to write temporary files in `/` and fails with permission denied, which make zone transfers fail.

Example logs:
```
Nov 29 12:42:51 sh-hn03 named[41433]: zone 9.10.IN-ADDR.ARPA/IN: Transfer started.
Nov 29 12:42:51 sh-hn03 named[41433]: transfer of '9.10.IN-ADDR.ARPA/IN' from 10.10.0.1#53: connected using 10.10.0.3#47676
Nov 29 12:42:51 sh-hn03 named[41433]: zone 9.10.IN-ADDR.ARPA/IN: transferred serial 2017082427
Nov 29 12:42:51 sh-hn03 named[41433]: transfer of '9.10.IN-ADDR.ARPA/IN' from 10.10.0.1#53: Transfer completed: 1 messages, 353 records, 10967 bytes, 0.004 secs (2741750 bytes/sec)
Nov 29 12:42:51 sh-hn03 named[41433]: dumping master file: tmp-iF1hWGAVRI: open: permission denied
```

The `directory "/var/named";` directive is properly added in `/etc/named.conf` on the master server, but for some reason is absent on DNS slaves.

**Comment by kcgthb** [comment]
According to the Bind9 documentation at https://ftp.isc.org/isc/bind9/cur/9.9/doc/arm/Bv9ARM.ch06.html

> **directory**
 The working directory of the server. Any non-absolute pathnames in the configuration file will be taken as relative to this directory. The default location for most server output files (e.g. named.run) is this directory. If a directory is not specified, the working directory defaults to `.', the directory from which the server was started. The directory specified should be an absolute path.

So without that option, the working directory for `named` will default to the directory from where the service was started, ie. `/`.

**Comment by kcgthb** [comment]
Tentative fix in PR #4393. 
I'm not entirely sure to understand the rationale behind the original test, but this seem to work in my case.

**Comment by bybai** [comment]
Hi @kcgthb, thanks your PR.
